bridgerelayer/chain/substrate/chain.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Snowfork
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package substrate
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/sirupsen/logrus"
+	"github.com/snowfork/polkadot-ethereum/bridgerelayer/chain"
+	"github.com/snowfork/polkadot-ethereum/bridgerelayer/crypto/sr25519"
+	"github.com/spf13/viper"
+)
+
+type Chain struct {
+	listener *Listener
+	writer   *Writer
+	conn     *Connection
+}
+
+const Name = "Substrate"
+
+func NewChain(ethMessages chan chain.Message, subMessages chan chain.Message) (*Chain, error) {
```

Comment:
Can we be more specific with argument names and channel types here? 

```suggestion
func NewChain(msgsToSend <-chan chain.Message, msgsObserved chan<- chain.Message) (*Chain, error) {
```

Reply:
Added #101 to track this.